src/overlays/actors/ovl_Bg_Toki_Hikari/z_bg_toki_hikari.c

```diff
@@ -28,23 +40,152 @@ const ActorInit Bg_Toki_Hikari_InitVars = {
+}
+
+void BgTokiHikari_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgTokiHikari* this = THIS;
+    this->actionFunc(this, globalCtx);
+}
+
+void BgTokiHikari_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    BgTokiHikari* this = THIS;
+
+    switch (this->actor.params) {
+        case 0:
+            func_808BA018(this, globalCtx);
+            break;
+        case 1:
+            func_808BA2CC(this, globalCtx);
+            break;
+    }
+}
+
+void func_808BA018(BgTokiHikari* this, GlobalContext* globalCtx) {
+    s32 pad;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_toki_hikari.c", 246);
+    func_80093D18(globalCtx->state.gfxCtx);
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_toki_hikari.c", 252),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+    if (gSaveContext.linkAge == 0) {
+        gSPDisplayList(POLY_OPA_DISP++, D_06008190);
+    } else {
+        gSPDisplayList(POLY_OPA_DISP++, D_06007E20);
+        func_80093D84(globalCtx->state.gfxCtx);
+        gSPSegment(POLY_XLU_DISP++, 8,
+                   Gfx_TexScroll(globalCtx->state.gfxCtx, 0, globalCtx->gameplayFrames & 127, 64, 32));
+
+        gSPSegment(POLY_XLU_DISP++, 9,
+                   Gfx_TexScroll(globalCtx->state.gfxCtx, 0, globalCtx->gameplayFrames & 127, 64, 32));
+
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_toki_hikari.c", 278),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+        gSPDisplayList(POLY_XLU_DISP++, D_06007EE0);
+    }
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_bg_toki_hikari.c", 284);
+}
+
+void func_808BA204(BgTokiHikari* this, GlobalContext* globalCtx) {
+    if (globalCtx->unk_11D30[1] != 0) {
+        this->actionFunc = func_808BA22C;
+    }
+}
+
+void func_808BA22C(BgTokiHikari* this, GlobalContext* globalCtx) {
+    if (this->unk_14C < 1.0f) {
+        this->unk_14C += 0.05f;
+    } else {
+        this->unk_14C = 1.0f;
+        this->actionFunc = func_808BA274;
+    }
+}
+
+void func_808BA274(BgTokiHikari* this, GlobalContext* globalCtx) {
+    if (this->unk_14C > 0.2f) {
+        this->unk_14C -= 0.025f;
+    } else {
+        this->unk_14C = 0.0f;
+        Actor_Kill(&this->actor);
+    }
+}
+
+void func_808BA2CC(BgTokiHikari* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_toki_hikari.c", 0x15E);
+    Matrix_Translate(0.0f, 276.0f, 1122.0f, 0);
+    Matrix_Scale(0.32f, 0.32f, this->unk_14C * 7.0f, 1);
+    Matrix_RotateZ(M_PI, 1);
+    func_80093D18(globalCtx->state.gfxCtx);
+    Matrix_Push();
+    gDPPipeSync(POLY_XLU_DISP++);
+    gDPSetPrimColor(POLY_XLU_DISP++, 0, 0x80, (this->unk_14C * 255.0f), ((u8)(155.0f * this->unk_14C) + 0x64),
+                    (this->unk_14C * 255.0f), (this->unk_14C * 255.0f));
+
+    gDPSetEnvColor(POLY_XLU_DISP++, ((u8)(this->unk_14C * 155.0f) + 0x64), (u8)(255.0f * this->unk_14C), 0, 128);
```

Comment:
```suggestion
    gDPSetPrimColor(POLY_XLU_DISP++, 0, 0x80, this->unk_14C * 255.0f, (u8)(155.0f * this->unk_14C) + 100,
                    this->unk_14C * 255.0f, this->unk_14C * 255.0f);

    gDPSetEnvColor(POLY_XLU_DISP++, (u8)(this->unk_14C * 155.0f) + 100, this->unk_14C * 255.0f, 0, 128);
```
Makes more sense in decimal, and the parentheses are unneeded. Can SetEnvColor be written like this and still match? If not, does it work to have `(u8)` casts on all the other products?

Reply:
It still needs to have the (u8) cast to work even with the change :/

Reply:
what makes it not match anymore? flipping the order of the adds?

Reply:
Having the `(u8)(255.0f * this->unk_14C)` will match, if I try the suggested `this->unk_14C * 255.0f` it causes the mismatch. I tried changing the other instances of having `this->unk_14C * <float_num>` too to see if that was the issue, but it did not end up matching.

Reply:
so just flipping the mult causes the non match, even if you leave the cast there?